src/main/java/com/palantir/docker/compose/execution/ConflictingContainerRemovingDockerCompose.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.docker.compose.execution;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConflictingContainerRemovingDockerCompose extends DelegatingDockerCompose {
+    private static final Logger log = LoggerFactory.getLogger(ConflictingContainerRemovingDockerCompose.class);
+    private static final Pattern NAME_CONFLICT_PATTERN = Pattern.compile("The name \"([^\"]*)\" is already in use");
+
+    private final Docker docker;
+    private final int retryAttempts;
+
+    public ConflictingContainerRemovingDockerCompose(DockerCompose dockerCompose, Docker docker) {
+        this(dockerCompose, docker, 1);
+    }
+
+    public ConflictingContainerRemovingDockerCompose(DockerCompose dockerCompose, Docker docker, int retryAttempts) {
+        super(dockerCompose);
+
+        Preconditions.checkArgument(retryAttempts >= 1, "retryAttempts must be at least 1, was " + retryAttempts);
+        this.docker = docker;
+        this.retryAttempts = retryAttempts;
+    }
+
+    @Override
+    public void up() throws IOException, InterruptedException {
+        int currRetryAttempt = 0;
+
+        while (true) {
+            try {
+                super.up();
```

Comment:
This is pretty nitty, but it seems odd to override a method that you can't then actually reimplement yourself. How about making the delegated docker composition protected rather than private?